lib/ansible/module_utils/facts/system/pkg_mgr.py

```diff
@@ -68,6 +68,20 @@ class PkgMgrFactCollector(BaseFactCollector):
     name = 'pkg_mgr'
     _fact_ids = set()
     _platform = 'Generic'
+    required_facts = set(['distribution'])
+
+    def _check_rh_versions(self, collected_facts):
+        if collected_facts['ansible_distribution'] == 'Fedora':
```

Comment:
Maybe we should also check whether yum or dnf exists in the PATH before setting it in this function?